File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/DocsRecentView.java
package app.editors.manager.mvp.views.main;

import android.net.Uri;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import java.util.List;

import app.editors.manager.mvp.models.account.Recent;
import app.editors.manager.mvp.models.base.Entity;
import app.editors.manager.mvp.models.explorer.Explorer;
import app.editors.manager.mvp.models.explorer.File;
import app.editors.manager.mvp.views.base.BaseView;
import app.editors.manager.ui.dialogs.ContextBottomDialog;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface DocsRecentView extends DocsBaseView {

    void updateFiles(List<Entity> lastModifiedFiles);
    void openFile(File response);
    void onMoveElement(Recent recent, int position);
    void onContextShow(ContextBottomDialog.State state);
    void onDeleteItem(int mContextPosition);
    void onOpenDocs(Uri uri);
    void onOpenCells(Uri uri);
    void onOpenPresentation(Uri uri);
    void onOpenPdf(Uri uri);
    void onOpenMedia(Explorer images, boolean isWebDav);
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/DocsRecentView.kt
package app.editors.manager.mvp.views.main

import app.documents.core.network.manager.models.explorer.CloudFile
import app.documents.core.storage.recent.Recent
import app.editors.manager.mvp.presenters.main.OpenState
import app.editors.manager.mvp.presenters.main.RecentState
import lib.toolkit.base.OpenMode
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface DocsRecentView : DocsBaseView {
    fun updateFiles(files: List<Recent>, sortByUpdated: Boolean)
    fun openFile(response: CloudFile, openMode: OpenMode)
    fun onDeleteItem(position: Int)
    fun onRender(state: RecentState)
    fun onOpenFile(state: OpenState)
}